### ISSUE TITLE
Tool Call Observer 

### ISSUE THREAD

**Issue by 714Lemons** [FlowiseAI/FlowiseChatEmbed#222]
It would be awesome to have an observer to listen for tool calls in the frontend to trigger actions. A separate observer and not just an include in the messages array would be great to listen for new actions without additional logic.

**Comment by neehhaa06** [comment]
Hello @714Lemons 

Maybe consider implementing an observer pattern to manage tool calls, where an observer interface and manager class handle notifications for various actions. This approach separates tool logic from action handling, simplifying both management and extension. Integrate this pattern into your front end to efficiently respond to tool actions with minimal additional logic.

**Comment by 714Lemons** [comment]
Yes i am already doing it like that but its very hacky and a consistent solution for this would be great.
